src/main/java/com/yelp/nrtsearch/server/plugins/CustomRequestPlugin.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.plugins;
+
+import java.util.Map;
+
+/**
+ * Plugin interface that allows adding one or more routes and their corresponding processors which
+ * can be called via the `custom` RPC. This can be implemented by plugins to modify behavior without
+ * requiring updating config files and restarting nrtsearch.
+ */
+public interface CustomRequestPlugin {
+
+  @FunctionalInterface
+  interface RequestProcessor {
+    Map<String, String> process(Map<String, String> request);
```

Comment:
Should the path also be passed as a parameter? It would give the plugin more flexibility with its processor implementations.

Reply:
I was expecting a RequestProcessor per path, but I guess providing the path is not a huge overhead and it does provide more flexibility. I'll add it.